test/index.js

```diff
@@ -0,0 +1,160 @@
+var events = require('events');
+var path = require('path');
+var quickfix = require('../');
+var should = require('should');
+
+var acceptorPropertiesFile = path.resolve(path.join(__dirname, 'support', 'acceptor.properties'));
+var initiatorPropertiesFile = path.resolve(path.join(__dirname, 'support', 'initiator.properties'));
+
+describe('node-quickfix', function () {
+
+  describe('initiator', function () {
+  
+    it('should throw if not supplied options', function () {
+      (function () { quickfix.initiator(); }).should.throw('FixInitiator requires an options parameter');
+    });
+
+    it('should throw if supplied options with no settings or propertiesFile', function () {
+      (function () { quickfix.initiator({}, {}); }).should.throw('you must provide FixInitiator either an options.settings string or options.propertiesFile path to a properties file');
+    });
+
+    it('should create a new initiator instance with fix beginString, senderCompID, targetCompID on defined session', function () {
+      var initiator = new quickfix.initiator({}, {
+        propertiesFile: initiatorPropertiesFile
+      });
+      initiator.should.not.eql(null);
+      initiator.isLoggedOn().should.eql(false);
+      initiator.getSessions()[0].beginString.should.eql('FIX.4.4');
+      initiator.getSessions()[0].senderCompID.should.eql('NODEQUICKFIX');
+      initiator.getSessions()[0].targetCompID.should.eql('ELECTRONIFIE');
+    });
+
+    it('should start and stop when called', function (done) {
+      var initiator = new quickfix.initiator({}, {
+        propertiesFile: initiatorPropertiesFile
+      });
+      initiator.start(function () {
+        // todo: started with local property
+        initiator.stop(function () {
+          // todo: started with local property
+          done();  
+        });
+      });
+    });
+
+  });
+
+  describe('acceptor', function () {
+  
+    it('should throw if not supplied options', function () {
+      (function () { quickfix.acceptor(); }).should.throw('FixAcceptor requires an options parameter');
+    });
+
+    it('should throw if supplied options with no settings or propertiesFile', function () {
+      (function () { quickfix.acceptor({}, {}); }).should.throw('you must provide FixAcceptor either an options.settings string or options.propertiesFile path to a properties file');
+    });
+
+    it('should create a new acceptor instance with fix beginString, senderCompID, targetCompID on defined session', function () {
+      var acceptor = new quickfix.acceptor({}, {
+        propertiesFile: acceptorPropertiesFile
+      });
+
+      acceptor.should.not.eql(null);
+      acceptor.getSessions()[0].beginString.should.eql('FIX.4.4');
+      acceptor.getSessions()[0].senderCompID.should.eql('ELECTRONIFIE');
```

Comment:
I've changed these to TargetCompID and SenderCompID in my continuation of this branch. Looks like this just got merged, but it'll change to be more generic with the next PR.